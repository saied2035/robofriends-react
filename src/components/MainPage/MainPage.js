import {useEffect,memo} from 'react';
import  CardList from '../cardList/cardList';
import SearchBox from '../searchBox/searchBox';
import Header from '../Header/Header';
import './MainPage.css';
const MainPage=(props) => {
    const {onRequestRobots,onSearchChange,filterRobots,isPending,error,robots} = props

    useEffect(() =>{
      onRequestRobots()
    },[onRequestRobots])

    return (
               isPending ? 
                   <div style={{height:'100vh'}}className='flex justify-center items-center'><h1>Loading</h1></div>
               :
               (
                  <div className="tc">
                  <Header/>
                  <SearchBox onSearchChange={(event) => onSearchChange(event,robots)}/>
                  <CardList robots={filterRobots} error={error}/>    
                  </div>
               )
    );
}

export default memo(MainPage);