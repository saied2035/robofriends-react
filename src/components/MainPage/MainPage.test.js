import {shallow} from 'enzyme'
import MainPage from './MainPage'
describe("MainPage",() => {
    const onSearchChange= jest.fn()
    const onRequestRobots= jest.fn()
    const error = ""
    let robots=[]
    let filterRobots= []
    let isPending = true
    const props1 = {onSearchChange,error,filterRobots,robots,isPending,onRequestRobots}
    robots = [
        {
            id : 1,
            name: 'saied',
            email: 'saied@gmail.com'
        },
        {
            id : 2,
            name: 'saied1',
            email: 'saied1@gmail.com'
        },
        {
            id : 3,
            name: 'saied2',
            email: 'saied2@gmail.com'
        },
        {
            id : 4,
            name: 'saied3',
            email: 'saied3@gmail.com'
        }
    ]
    filterRobots = robots
    isPending= false
    const props2 = {onSearchChange,error,filterRobots,robots,isPending,onRequestRobots}    
    const wrapperMainPage1 = shallow(<MainPage {...props1}/>)
    const wrapperMainPage2 = shallow(<MainPage {...props2}/>)

        it("render MainPage pending case",() => {
            expect(wrapperMainPage1).toMatchSnapshot()
        })
        it("render MainPage no pending case",() => {
            expect(wrapperMainPage2).toMatchSnapshot()
        })

        it("try call filterrobots", () => {        
                const event ={
                    target: {
                        value: 'test'
                    }
                }
                wrapperMainPage2.find('Memo(SearchBox)').props().onSearchChange(event,robots)
                expect(onSearchChange).toBeCalledWith(event,robots)
        })
                
})