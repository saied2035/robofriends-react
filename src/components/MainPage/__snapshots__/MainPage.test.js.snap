// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage render MainPage no pending case 1`] = `
<div
  className="tc"
>
  <Memo(Header) />
  <Memo(SearchBox)
    onSearchChange={[Function]}
  />
  <Memo(CardList)
    error=""
    robots={
      Array [
        Object {
          "email": "saied@gmail.com",
          "id": 1,
          "name": "saied",
        },
        Object {
          "email": "saied1@gmail.com",
          "id": 2,
          "name": "saied1",
        },
        Object {
          "email": "saied2@gmail.com",
          "id": 3,
          "name": "saied2",
        },
        Object {
          "email": "saied3@gmail.com",
          "id": 4,
          "name": "saied3",
        },
      ]
    }
  />
</div>
`;

exports[`MainPage render MainPage pending case 1`] = `
<div
  className="flex justify-center items-center"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <h1>
    Loading
  </h1>
</div>
`;
